File: easy/0836_rectangle_overlap/solution.cpp
#include <vector>

class Solution {
public:
  bool isRectangleOverlap(std::vector<int> &rec1, std::vector<int> &rec2) {
    int x11 = rec1[0], y11 = rec1[1], x12 = rec1[2], y12 = rec1[3];
    int x21 = rec2[0], y21 = rec2[1], x22 = rec2[2], y22 = rec2[3];

    return !(x11 >= x22 || x12 <= x21 || y11 >= y22 || y12 <= y21);
  }
};

File: easy/0836_rectangle_overlap/solution.py
from typing import List


class Solution:
    def isRectangleOverlap(self, rec1: List[int], rec2: List[int]) -> bool:
        x11, y11, x12, y12 = rec1
        x21, y21, x22, y22 = rec2

        return not (x11 >= x22 or x12 <= x21 or y11 >= y22 or y12 <= y21)
